_security/configuration/index.md

```diff
@@ -14,18 +14,74 @@ redirect_from:
+  
+- Steps:
+  - Generate your own certificates: Use tools like OpenSSL or a certificate authority (CA) to generate your own certificates.
+  - Place generated certificates and private key to the appropriate OpenSearch directory.
+  - Update file permissions: Ensure the files are accessible by the OpenSearch service user and have proper permissions set to avoid unauthorized access.
+
+2. [Reconfigure `opensearch.yml` to use your certificates]({{site.url}}{{site.baseurl}}/security/configuration/tls). 
+- The `opensearch.yml` file is the main configuration file for OpenSearch. Update this file to point to your custom certificates for secure communication.
+- Steps:
+  - Open `opensearch.yml`: Locate and open the `opensearch.yml` file in your preferred text editor.
+  - Update paths to certificates and keys: Set the correct paths for your certificates and keys in the `opensearch.yml` file. For example:
+```json
+plugins.security.ssl.transport.pemcert_filepath: /path/to/your/cert.pem
+plugins.security.ssl.transport.pemkey_filepath: /path/to/your/key.pem
+plugins.security.ssl.transport.pemtrustedcas_filepath: /path/to/your/ca.pem
+plugins.security.ssl.http.enabled: true
+plugins.security.ssl.http.pemcert_filepath: /path/to/your/cert.pem
+plugins.security.ssl.http.pemkey_filepath: /path/to/your/key.pem
+plugins.security.ssl.http.pemtrustedcas_filepath: /path/to/your/ca.pem
+```
+3. [Reconfigure `config.yml` to use your authentication backend]({{site.url}}{{site.baseurl}}/security/configuration/configuration/) (if you don't plan to use the internal user database).
+- The `config.yml` file allows you to configure the authentication and authorization mechanisms for OpenSearch.
+- Update the authentication backend settings in `config/opensearch-security/config.yml` according to your requirements. For example, to use LDAP:
+    ```json
+    authc:
+    ldap:
+    http_enabled: true
+    transport_enabled: true
+    order: 1
+    http_authenticator:
+    type: basic
+    challenge: true
+    authentication_backend:
+    type: ldap
+    config:
+    ```
+  
+4. [Modify the configuration YAML files]({{site.url}}{{site.baseurl}}/security/configuration/yaml).
+
+Determine which additional YAML files need modification, for example the `roles.yml`, `roles_mapping.yml`, `internal_users.yml`, etc.
+Edit the files: Open each required file in your preferred text editor and make the necessary adjustments. For example, adding roles or updating role mappings:
+
+5. If you plan to use the internal user database, [set a password policy in `opensearch.yml`]({{site.url}}{{site.baseurl}}/security/configuration/yaml/#opensearchyml).
+- If you opt to use the internal user database for authentication, it is advisable to enforce a password policy to ensure strong passwords.
+  1. Open `opensearch.yml` and add or update the password policy settings.
+
+6. [Apply changes using the `securityadmin` script]({{site.url}}{{site.baseurl}}/security/configuration/security-admin).
```

Comment:
This may be confusing to some first time users. In the demo configuration, the security index is automatically sourced from the yaml files and securityadmin is only needed to apply changes. 

If a user sets `plugins.security.allow_default_init_securityindex` to `false` then it would be required to run securityadmin since that setting prevents the security index to initialize from yaml files by default.